controllers/toolchaincluster/toolchaincluster_controller_test.go

```diff
@@ -119,6 +124,53 @@ func TestClusterControllerChecks(t *testing.T) {
 		require.NoError(t, err)
 		assertClusterStatus(t, cl, "stable", offline())
 	})
+
+	t.Run("migrates connection settings to kubeconfig in secret", func(t *testing.T) {
+		// given
+		tc, secret := newToolchainCluster("tc", tcNs, "http://cluster.com", toolchainv1alpha1.ToolchainClusterStatus{})
+		cl := test.NewFakeClient(t, tc, secret)
+		reset := setupCachedClusters(t, cl, tc)
+		defer reset()
+
+		controller, req := prepareReconcile(tc, cl, requeAfter)
+		expectedKubeConfig := composeKubeConfigFromData([]byte("mycooltoken"), "http://cluster.com", "test-namespace", true)
```

Comment:
should we use a fake/hardcoded kubeconfig ? If there's a bug in the `composeKubeConfigFromData` function this test will not catch it right , since both the reconciler and test are using the same logic for generating the kubeconfig.

Reply:
maybe the `TestComposeKubeConfig` would catch it.

Reply:
The `composeKubeConfigFromData` function is really a glorified struct initialization, so I chose to use that. Having a hardcoded config with all the `Extension` fields would IMHO require the same amount of explanation as the customized DeepEqual. So I chose this way that I think is ever so slightly more legible.

And you're right, the `TestComposeKubeConfig` function is there for checking the validity of the generated kubeconfig, so I think it is OK to have 1 function used both in test and the tested reconcile (if it becomes more complicated, I would agree with you that it could cause a problem).

Reply:
yup not a big problem! 

I was thinking something similar to https://github.com/codeready-toolchain/host-operator/blob/43bf61e3a43ca2444883348059e0fe3d3177dff3/controllers/spacerequest/spacerequest_controller_test.go#L1094 for generating a fake kubeconfig based on what you expect the actual one should look like. I'm not sure I understand which `Extensions` fields you are referring too, but I'm sure I'm missing something here.